internal/tags/tags.go

```diff
@@ -0,0 +1,64 @@
+package tags
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/trento-project/trento/internal/consul"
+)
+
+const kvTagsPath string = "trento/v0/tags/%s/%s/"
+
+type Tags struct {
+	client   consul.Client
+	resource string
+	id       string
+}
+
+func NewTags(client consul.Client, resource string, id string) *Tags {
+	return &Tags{
+		client:   client,
+		resource: resource,
+		id:       id,
+	}
+}
+
+func (t *Tags) getKvTagsPath() string {
+	return fmt.Sprintf(kvTagsPath, t.resource, t.id)
+}
+
+func (t *Tags) GetAll() ([]string, error) {
+	path := t.getKvTagsPath()
+
+	tagsMap, err := t.client.KV().ListMap(path, path)
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving tags")
+	}
+
+	var tags []string
+	for tag := range tagsMap {
+		tags = append(tags, tag)
+	}
+
+	return tags, nil
+}
+
+func (t *Tags) Create(tag string) error {
+	path := t.getKvTagsPath() + tag + "/"
+
+	if err := t.client.KV().PutMap(path, nil); err != nil {
```

Comment:
I don't really get it here. You are posting a map without values, and the real value comes in the path itself. Shouldn't we have better a key/value format and put the tag as value?